torchbenchmark/models/pplbench_beanmachine/__init__.py

```diff
@@ -20,6 +20,9 @@ def __init__(self, test):
 
         # Get data for inference & evaluating model
         if test == "eval":
+            # The evaluation step doesn't use PyTorch code because it needs to handle
+            # various probabilistic programming languages. See GitHub issue #898
+            raise NotImplementedError("PPLBench Beanmachine eval test doesn't use PyTorch and is disabled - see GH issue #898.")
```

Comment:
The `get_module()` API just returns self.model which is set on line 19.

That will never use PyTorch.

Reply:
We should refactor the benchmark not to measure the non-PyTorch regression.

Reply:
I updated the code, sorry about the bug. I want to clarify that the `get_module()` API returns the model set on line [76](https://github.com/pytorch/benchmark/blob/01d9cd9209eee91ca9f3c9c9eb90b8dfabf8f109/torchbenchmark/models/pplbench_beanmachine/__init__.py#L76), not line 19.

Reply:
We are still passing the `robust_regression.RobustRegression` to the model on line 30.

How much of the time of this mode is spent in non-PyTorch code?

Reply:
> We are still passing the `robust_regression.RobustRegression` to the model on line 30.

The naming is getting confusing :)... The `robust_regression.RobustRegression` class on line 30 is not the same as xarray data generation class:

In `pplbench.models`, there's a [robust regression class](https://github.com/facebookresearch/pplbench/blob/main/pplbench/models/robust_regression.py) that's used for data generation and evaluation.

In `pplbench.ppls.beanmachine`, there's also a [robust regression class](https://github.com/facebookresearch/pplbench/blob/main/pplbench/ppls/beanmachine/robust_regression.py#L18), which defines the model in Bean Machine (with pytorch distributions). We are passing this class to MCMC inference method so it knows which model to run when we call `infer`.

Reply:
@horizon-blue so to confirm, xarray is *not* used and 100% of the time is spent in pytorch?

Reply:
*Most* of the operations in  [`forward`](https://github.com/pytorch/benchmark/blob/97efd8b0dbec4aaae5f95a76dfee97f6859556e6/torchbenchmark/models/pplbench_beanmachine/__init__.py#L46-L47) is done in pytorch. xarray is still used to parse and convert the data.

You mentioned in the other thread that TorchDynamo only calls `get_module()`, in which case `forward` wouldn't be invoked anyways, which is probably why you aren't seeing torch operator in your profiler.

:) I begin to feel like it might be easier if we rework this benchmark to use Bean Machine inference library directly instead of via PPL Bench. It doesn't make sense to introduce an extra layer of complexity when we only want to measure the core computation.

Reply:
> _Most_ of the operations in [`forward`](https://github.com/pytorch/benchmark/blob/97efd8b0dbec4aaae5f95a76dfee97f6859556e6/torchbenchmark/models/pplbench_beanmachine/__init__.py#L46-L47) is done in pytorch. xarray is still used to parse and convert the data.

Can we quantify how much time is spent in PyTorch, versus other libraries?  Perhaps using the PyTorch profiler.